service/build.go

```diff
@@ -379,3 +413,23 @@ func (uis *UIServer) buildHistory(w http.ResponseWriter, r *http.Request) {
 
 	gimlet.WriteJSON(w, history)
 }
+
+func getBuildTaskMap(builds []build.Build) (map[string]map[string]task.Task, error) {
+	buildIds := make([]string, 0, len(builds))
+	for _, b := range builds {
+		buildIds = append(buildIds, b.Id)
+	}
+	tasksForBuilds, err := task.FindAll(task.ByBuildIds(buildIds).WithFields(task.BuildIdKey, task.DisplayNameKey, task.StatusKey, task.DetailsKey))
+	if err != nil {
+		return nil, errors.Wrap(err, "can't get tasks for builds")
+	}
+	buildTaskMap := make(map[string]map[string]task.Task)
```

Comment:
can you add some clarity comments about this data structure? Possibly a private type would be more readable, with two maps map[buildId][]string{taskIds} and map[taskId]task.Task{} but it is private so either does work

Reply:
Splitting into two maps makes it much clearer.
I opted not to define a type.